sdk/tests/conformance/textures/misc/tex-image-with-bad-args.html

```diff
@@ -0,0 +1,101 @@
+** EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+** MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+** IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+** CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+** TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+** MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
+*/
+
+-->
+<!DOCTYPE html>
+<html>
+<head>
+<meta charset="utf-8">
+<link rel="stylesheet" href="../../../resources/js-test-style.css"/>
+<script src="../../../js/js-test-pre.js"></script>
+<script src="../../../js/webgl-test-utils.js"></script>
+</head>
+<body>
+<canvas id="c" width="16" height="16"></canvas>
+<div id="description"></div>
+<div id="console"></div>
+<script>
+"use strict";
+description('Tests texImage2D with invalid internalformat/format/type combinations');
+debug("This is a regression test for <a href='https://bugs.chromium.org/p/chromium/issues/detail?id=656889'>Chromium Issue 656889</a>");
+
+var wtu = WebGLTestUtils;
+var contextVersion = wtu.getDefault3DContextVersion();
+var gl = wtu.create3DContext("c");
+var ext = null;
+if (!gl) {
+    testFailed("WebGL context does not exist");
+} else {
+    testPassed("WebGL context exists");
+}
+
+var doTexImage = function(test) {
+    var tex = gl.createTexture();
+    gl.bindTexture(gl.TEXTURE_2D, tex);
+    gl.texImage2D(gl.TEXTURE_2D, 0, test.internalformat, 64, 64, 0, test.format, test.type, null);
+    wtu.glErrorShouldBe(gl, test.errors, "TexImage2D taking " + wtu.glEnumToString(gl, test.internalformat) + "/" +
+        wtu.glEnumToString(gl, test.format) + "/" + wtu.glEnumToString(gl, test.type));
+    gl.deleteTexture(tex);
+}
+
+var tests = [
+    { internalformat: gl.RGBA, format: gl.RGBA, type: gl.UNSIGNED_BYTE, errors: [gl.NO_ERROR] },
+];
+
+if (contextVersion < 2) {
+    var webgl1Tests = [
+        { internalformat: gl.RGBA, format: gl.RGBA, type: gl.FLOAT, errors: [gl.INVALID_OPERATION, gl.INVALID_ENUM] },
+    ];
+    webgl1Tests.concat(tests);
+    webgl1Tests.forEach(doTexImage);
+
+    ext = gl.getExtension('OES_texture_float');
+    if (ext) {
+        var extensionTests = [
+            { internalformat: gl.RGBA, format: gl.RGBA, type: gl.FLOAT, errors: [gl.NO_ERROR] },
+        ];
+        extensionTests.forEach(doTexImage);
+    }
+    ext = gl.getExtension('OES_texture_half_float');
+    if (ext) {
+        var extensionTests = [
+            { internalformat: gl.RGBA, format: gl.RGBA, type: ext.HALF_FLOAT_OES, errors: [gl.NO_ERROR] },
+        ];
+        extensionTests.forEach(doTexImage);
+    }
+} else {
+    var webgl2Tests = [
+        { internalformat: gl.RGBA, format: gl.RGBA, type: gl.FLOAT, errors: [gl.INVALID_OPERATION] },
+        { internalformat: gl.RGBA, format: gl.RGBA, type: gl.HALF_FLOAT, errors: [gl.INVALID_OPERATION] },
+    ];
+    webgl2Tests.concat(tests);
+    webgl2Tests.forEach(doTexImage);
+}
```

Comment:
WebGL1 test coverage is already in the extension tests.

Why don't we only keep the WebGL2 test cases and move to conformance2/textures.misc?

Also, please test LUMINANCE as well because someone just reported using float luminance textures successfully in WebGL2 (which should be disallowed)